### ISSUE TITLE
Getting an error message after putting in my Steam ID

### ISSUE THREAD

**Issue by Tagswell** [cristan2/steam-wishlist-demos#2]
![demo wishlist error](https://github.com/user-attachments/assets/5669f3e2-b7f7-430e-83e1-4d46b6b7c210)

Getting this error message after copying in my Steam ID.
I'm not sure what it means exactly, or what the issue causing it is.
I've used the site with no issues before so this is a new issue for me.

Hope this isn't a repeat issue, user error, or something silly on my end!

**Comment by cristan2** [comment]
Have you checked if your wishlist is set to public in your privacy settings?

In any case, Steam made some API changes and the demo info is no longer available as before, so the app doesn't work properly anyway at the moment. I'm working on a fix, hoping to release an update in time for the winter sale.

**Comment by Tagswell** [comment]
Yes I did double check that and the only thing that was private is my
friends list.

Happy to help with my issue however I can, and looking forward to the fix
as well!

Thanks for responding and letting me know that it's not functional at the
moment anyways.

**Comment by cristan2** [comment]
Just pushed an update with a quick fix, could you try it out? Thanks.

There will still be some gaps in the available games for now, but at least everything should work faster. And I'm working on a few more improvements for the next weeks.

**Comment by Tagswell** [comment]
Hey, just got around to checking it again. Now works just as it did before!

Thanks again for responding and working on this project, it's very helpful.
= )
Hope you enjoyed the holidays and the new year, looking forward to seeing
what improvements you have planned!

**Comment by cristan2** [comment]
Glad you find it useful. Happy new year!

**Comment by gweronimo** [comment]
> Just pushed an update with a quick fix, could you try it out? Thanks.

@cristan2 - Looks like this quick-fix was not committed (pushed) to the code in this repo? (It works fine on the published webpage, but the code in the GitHub-repo seems to be an older version.)

For reference, the issue was likely that the `https://store.steampowered.com/wishlist/profiles/{steamid}/wishlistdata?p={page}` endpoint no longer works. The following API call can be used instead, but returns only `appid`, `priority` and `date_added` for each wishlist item:
```
resp = requests.get(
    f"https://api.steampowered.com/IWishlistService/GetWishlist/v1/",
    params={"steamid": steamid},
)
print(resp.json())
```

Any chance the fixed code could be pushed to the repo? I'm trying to build a local tool with similar functionality, using the `FreeSimpleGUI` python-package, where I can also keep track of which demos I've already installed, tried, been unable to donwload, etc. Your implementation is great and has been very helpful, so I was a bit sad to discover that the repo-code was (apparently) not fully up-to-date...

**Comment by cristan2** [comment]
Hey man. Life got in the way, I haven't really been able to get back to this. There's a lot of new code I hadn't committed, it wasn't a simple fix. To get around the Steam API change, the solution I found was to prefetch all of Steam's games data and build a local sqlite database with it. Obviously, this is a very limited mirror of Steam's db, with just a few relevant info related to each game, including demo information. This is what your wishlist data is now checked against to see which game has a demo and that's why it's faster - no additional requests to steam. 

If you're working on your own solution, I've just pushed my code to a separate branch - [demo_indexing](https://github.com/cristan2/steam-wishlist-demos/tree/demo_indexing). If you have any use for it, go ahead, just keep in mind this is very much WIP, there's not much structure to it, the code is not clean, it's more like a proof of concept than anything. Basically, it's almost an entire reimplementation of the app and there are a lot of improvements to be made I didn't get around to do.

The basic concept is:
- read the list of appids from steam (there's an API for this)
- go through each appid and read its details from Steam; this has to be a new separate individual request, as this is the only way to get the demo information (no bulk endpoint for this)
- these requests have to be limited to 200 per 5 minutes (this is a Steam limitation, no way to get around this); I used some quick and dirty code for this, but there are libraries to do this in a nicer way (e.g. https://pypi.org/project/ratelimiter/)
- obviously, this will take a lot of time, last time I did this there were almost 200k entries. This was in December last year. You might want to limit your requests to only the last X entries or something. And there are many edge cases to address (you'll see some TODOs) and some solutions to mitigate this, but it takes time
- save all of this to an sqlite database

Then, when using the app, retrieving the wishlist is the only request that is sent to Steam in real time. Those games are checked against the local sqlite mirror. 

Hope this helps and good luck! I'll post an update here if and when I get back to this, I haven't given up on it, I still need this functionality too.

**Comment by gweronimo** [comment]
@cristan2 - Thank you very much for your elaborate answer, very helpful! I'll take a look at the new branch and the ratelimiter package. For my use case, I'll could cache the appids that are actually in my wishlist and update these (with rate-limiting) in the background... (But since demos come and go, I want the current state rather than a cached one.)
